build_tools/azure/posix.yml

```diff
@@ -37,11 +37,28 @@ jobs:
     TEST_DOCSTRINGS: 'false'
     CREATE_ISSUE_ON_TRACKER: 'false'
     SHOW_SHORT_SUMMARY: 'false'
+    COMMIT_MESSAGE: $[ dependencies.git_commit.outputs['commit.message'] ]
   strategy:
     matrix:
       ${{ insert }}: ${{ parameters.matrix }}
 
   steps:
+    - bash: |
+        set -ex
+        # Check if run only specific tests but on all random seeds
+        # commit message must take the form
+        # <title> [all random seeds]
+        # <test_name_1>
+        # <test_name_2>
+        # ...
+        if [[ "$COMMIT_MESSAGE" =~ \[all\ random\ seeds\] ]]; then
+          message=$(echo $COMMIT_MESSAGE | sed -n -e 's/^.*all\ random\ seeds\] //p')
+          selected="${message// / or }"
+        else
+          selected=""
+        fi
+        echo ##vso[task.setvariable variable=SELECTED_TESTS]$selected
+      displayName: Check selected tests for all random seeds
```

Comment:
Do we want this to only trigger for PRs?

```yaml
condition: eq(variables['Build.Reason'], 'PullRequest')
```

Reply:
I guess so